tests/test_windows.py

```diff
@@ -4,132 +4,99 @@
-                ((1, None), (1, None))).f == src.bounds.top - src.res[1]
-        assert src.window_transform(
-                ((-1, None), (-1, None))).c == src.bounds.left - src.res[0]
-        assert src.window_transform(
-                ((-1, None), (-1, None))).f == src.bounds.top + src.res[1]
+        with pytest.raises(ValueError):
+            # No height or width
+            from_bounds(left+EPS, bottom+EPS, right-EPS, top-EPS, src.transform)
 
 
 def test_window_transform_function():
     with rasterio.open('tests/data/RGB.byte.tif') as src:
-        transform = src.affine  # TODO transform
-        assert window_transform(transform, ((0, None), (0, None))) == src.affine
-        assert window_transform(transform, ((None, None), (None, None))) == src.affine
-        assert window_transform(
-                transform, ((1, None), (1, None))).c == src.bounds.left + src.res[0]
-        assert window_transform(
-                transform, ((1, None), (1, None))).f == src.bounds.top - src.res[1]
-        assert window_transform(
-                transform, ((-1, None), (-1, None))).c == src.bounds.left - src.res[0]
-        assert window_transform(
-                transform, ((-1, None), (-1, None))).f == src.bounds.top + src.res[1]
-
-
-
-def test_window_bounds_method():
-    with rasterio.open('tests/data/RGB.byte.tif') as src:
-        rows = src.height
-        cols = src.width
-        assert src.window_bounds(((0, rows), (0, cols))) == src.bounds
+        assert transform(((0, None), (0, None)), src.transform) == src.transform
+        assert transform(((None, None), (None, None)), src.transform) == src.transform
+        assert transform(
+            ((1, None), (1, None)), src.transform).c == src.bounds.left + src.res[0]
+        assert transform(
+            ((1, None), (1, None)), src.transform).f == src.bounds.top - src.res[1]
+        assert transform(
+            ((-1, None), (-1, None)), src.transform).c == src.bounds.left - src.res[0]
+        assert transform(
+            ((-1, None), (-1, None)), src.transform).f == src.bounds.top + src.res[1]
 
 
 def test_window_bounds_function():
     with rasterio.open('tests/data/RGB.byte.tif') as src:
         rows = src.height
         cols = src.width
-        transform = src.affine  # todo deprecate
-        assert window_bounds(transform, ((0, rows), (0, cols))) == src.bounds
+        assert bounds(((0, rows), (0, cols)), src.transform) == src.bounds
 
 
-def test_eval_window_bad_structure():
-    with pytest.raises(ValueError):
-        eval_window((1, 2, 3), 10, 10)
-    with pytest.raises(ValueError):
-        eval_window((1, 2), 10, 10)
+bad_windows = (
+    (1, 2, 3),
+    (1, 2),
+    ((1, 0), 2))
+
+@pytest.mark.parametrize("window", bad_windows)
+def test_eval_window_bad_structure(window):
     with pytest.raises(ValueError):
-        eval_window(((1, 0), 2), 10, 10)
+        evaluate(window, 10, 10)
 
 
 def test_eval_window_invalid_dims():
```

Comment:
@perrygeo you could probably use the above pytest parameterize annotation approach here too